diesel/src/pg/metadata_lookup.rs

```diff
@@ -47,7 +47,21 @@ where
                     .store_type(cache_key, type_metadata);
                 PgTypeMetadata(Ok(type_metadata))
             }
-            Err(_e) => PgTypeMetadata(Err(FailedToLookupTypeError::new(cache_key.into_owned()))),
+            Err(_e) => {
+                let metadata_cache = self.get_metadata_cache();
+                let lower_schema = schema.map(|schema| schema.to_lowercase()).unwrap();
+                if let Some(_) = metadata_cache.lookup_type(&PgMetadataCacheKey {
+                    schema: Some(Cow::Borrowed(lower_schema.as_str())),
+                    type_name: Cow::Borrowed(type_name.to_lowercase().as_str()),
+                }) {
```

Comment:
There is no need to use `Cow::Borrowed` here, as we don't need neither the lower cased type name nor the lower cased schema name later on. Constructing `PgMetadataCacheKey` can be simplified to:
```rust
PgMetadataCacheKey {
    schema: lower_schema.map(Cow::Owned),
    type_name: Cow::Owned(type_name.to_lowercase()),
}
```

`Cow` is some kind of "magic" type that either accepts a borrowed value (`&str` in this case) or the owned counter part (`String` in this case). It can be used to reduce unnecessary calls to clone. Generally speaking it should be only used in hot code paths, otherwise it mostly just complicate things.